### DISCUSSION TITLE
Feature request: Composer Frontline

### DISCUSSION THREAD

**Discussion by dg** [composer/composer#9588] [General]
I really miss the update command to update all (or some) dependencies to their latest version. I mean to update `composer.json` as well.

So a command that can update versions in `composer.json` from

```json
	"require": {
		"nette/application": "^2.4",
		"nette/bootstrap": "^3.0"
	},
```

to

```json
	"require": {
		"nette/application": "^3.1",
		"nette/bootstrap": "^3.1"
	},
```

There may be an option that decides whether to update within the version constaint (`^3.0 -> ^3.1`) or without (`^2.4 -> ^3.1`). And packages could be updated by mask in this way: (`frontline` is an example of a command name)

```
composer frontline  nette/*
```

Currently the same can be achieved by calling repeatedly `require`:

```
composer require nette/application
composer require nette/bootstrap
...
```

But it would be really useful to automate it.

I implemented it as a [script composer-frontline.php](https://gist.github.com/dg/734bebf55cf28ad6a5de1156d3099bff). I tried to implement it also as a [composer plugin](https://github.com/dg/composer-frontline), but I don't know internals enough so I throw in the towel.

**Comment by mfn** [comment]
Yarn has such a feature but it's much more => https://classic.yarnpkg.com/en/docs/cli/upgrade-interactive/

It's a godsend if you've LOTS of dependencies, and also the interactive part is useful:
![image](https://user-images.githubusercontent.com/87493/103540147-8d266a80-4e99-11eb-89e6-ae7bcd89f2dc.png)
